GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                        }
+
+                    }
+                    reports = await reports.GetNextPageAsync();
+                } while (reports != null);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting user's manager and reports {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            // group and role memberships
+            Console.WriteLine("\n Signed in user {0} is a member of the following Group and Roles (IDs)",
+                user.DisplayName);
+            IUserFetcher signedInUserFetcher = user;
+            try
+            {
+                IPagedCollection<IDirectoryObject> pagedCollection = await signedInUserFetcher.MemberOf.ExecuteAsync();
+                do
+                {
+                    List<IDirectoryObject> directoryObjects = pagedCollection.CurrentPage.ToList();
+                    foreach (IDirectoryObject directoryObject in directoryObjects)
+                    {
+                        if (directoryObject is Group)
+                        {
+                            Group group = directoryObject as Group;
+                            Console.WriteLine(" Group: {0}  Description: {1}", group.DisplayName, group.Description);
+                        }
+                        if (directoryObject is DirectoryRole)
+                        {
+                            DirectoryRole role = directoryObject as DirectoryRole;
+                            Console.WriteLine(" Role: {0}  Description: {1}", role.DisplayName, role.Description);
+                        }
+                    }
+                    pagedCollection = await pagedCollection.GetNextPageAsync();
+                } while (pagedCollection != null);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Signed in user's groups and roles memberships. {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            #endregion
+        }
+
+        private static async Task<User> CreateNewUser(IActiveDirectoryClient client, VerifiedDomain defaultDomain)
+        {
+            #region Create a new User
+
+            // **********************************************************
+            // Requires Directory.ReadWrite.All or Directory.AccessAsUser.All, and the signed in user
+            // must be a privileged user (like a company or user admin)
+            // **********************************************************
+
+            User newUser = new User();
+            if (defaultDomain.Name != null)
+            {
+                Console.WriteLine("\nCreating a new user...");
+                Console.WriteLine("\n  Please enter first name for new user:");
+                String firstName = Console.ReadLine();
```

Comment:
firstName = firstName.trim(); to remove the spaces surrounding the input parameter.

---
In reply to: [80769944](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80769944) [](ancestors = 80769944,80542793)